mempool/src/shared_mempool.rs

```diff
@@ -616,12 +622,20 @@ where
     }
 }
 
-fn process_config_update(config_update: OnChainConfigPayload) {
+async fn process_config_update<V>(smp: SharedMempool<V>, config_update: OnChainConfigPayload)
+where
+    V: TransactionValidation,
+{
     // check for VM update
-    if let Ok(_vm_publishing_option) = config_update.get::<VMPublishingOption>() {
-        // TODO restart VM validator
+    if config_update.get::<VMPublishingOption>().is_ok() {
```

Comment:
Wondering: do we anticipate config updates that do *not* require a restart? Most of the ones I can think of (publishing options, gas prices, max tx size, etc) would need a restart. Is it possible for us to write this in a way that it restarts on any config update (or any update except a blacklist)?

Reply:
Mempool receives config updates only when any of its subscribed configs (which as of now is just publishing options) have changed, so I can change this to restart on any config update

Can you explain why we wouldn't restart VM for a blacklist? not familiar with how this works (we're still talking about publishing options?

Reply:
Ah, sorry! I meant "a blacklist of config options that don't require restart" (that we would define), not publishing options.

Reply:
@sblackshear: I think you gave nice example. Gas price change doesn't require restart. It will be just reread from storage on a fly
I'm not confident that ratio will be in favor of restarting system components vs just reacting somehow to changes

Reply:
Ah, sorry I meant "gas cost", not "gas price". Gas cost change *does* require a restart because the gas cost table is only loaded when the VM starts up.

Reply:
Actually, it looks like all on-chain configs are loaded only on VM startup: https://github.com/libra/libra/blob/master/language/libra-vm/src/libra_vm.rs#L48. Am I reading this right @runtian-zhou?

Reply:
Yes. For VMValidator we will do a `LibraVM::new` and `LibraVM::load_configs` at startup time.

Reply:
Actually I would argue that we don't even need to restart VMValidator on a gas table change. The reason is that the VMValidator is running the Prologue anyway, which doesn't use on chain gas table at all.